Improve Readme (#85)
nanoframework/System.IO.Ports: README.md

Note that the port name **must** be `COMx` where x is a number. 

The `GetPortNames` method will give you a list of available ports:

```csharp
var ports = SerialPort.GetPortNames();
```

You can also directly specify the baud rate and other elements in the constructor:

```csharp
var port = new SerialPort("COM2", 115200);
```

Each property can be adjusted, including when the port is open. Be aware that this can generate hazardous behaviors. It is always recommended to change the properties while the port is closed.

**Important**: You should setup a timeout for the read and write operations. If you have none, read or a write periods may cause theads to be locked for indefinite periods.

```csharp
port.WriteTimeout = 1000;
port.ReadTimeout = 1000;
```

Note: some MCU do not support Handshake or specific bit parity even if you can set them up in the constructor.

### Opening and Closing the port

> The Serial Port can only operate once open and will finish the operations when closed. If you dispose the SerialPort, it will close it before some operations may have completed.

Reviewer: ```suggestion
The `SerialPort` can only operate once open and will finish the operations when closed. When disposed, the `SerialPort` will perform the close operation regardless of any ongoing receive or transmit operations.
```

### Read and Write

You have multiple functions to read and write, some are byte related, others string related. 

> Note that string functions will use the ASCII `Encoding` charset.

Reviewer: This is not correct. In nanoFramework UTF8 is the only supported encoding.

### Events

> #### Character

Reviewer: ```suggestion
#### Character

```

markdown rules ask for a empty line here

}
```

> #### WatchChar

Reviewer: ```suggestion
#### WatchChar

```

markdown rules ask for a empty line here
